java-client/src/main/java/com/wavefront/integrations/WavefrontDirectSender.java

```diff
@@ -0,0 +1,272 @@
+
+    final ApacheHttpClient4Engine apacheHttpClient4Engine = new ApacheHttpClient4Engine(httpClient, true);
+    apacheHttpClient4Engine.setFileUploadInMemoryThresholdLimit(100);
+    apacheHttpClient4Engine.setFileUploadMemoryUnit(ApacheHttpClient4Engine.MemoryUnit.MB);
+
+    ResteasyProviderFactory factory = ResteasyProviderFactory.getInstance();
+    ClientHttpEngine httpEngine = apacheHttpClient4Engine;
+
+    ResteasyClient client = new ResteasyClientBuilder().
+        httpEngine(httpEngine).
+        providerFactory(factory).
+        register(GZIPDecodingInterceptor.class).
+        register(AcceptEncodingGZIPFilter.class).
+        build();
+    ResteasyWebTarget target = client.target(server);
+    return target.proxy(DataIngestorAPI.class);
+  }
+
+  @Override
+  public synchronized void connect() throws IllegalStateException, IOException {
+    if (httpClient == null) {
+      directService = createWavefrontService(server, token);
+      scheduledFuture = scheduler.scheduleAtFixedRate(this, 1, 1, TimeUnit.SECONDS);
+    }
+  }
+
+  @Override
+  public void send(String name, double value) throws IOException {
+    addPoint(name, value, null, DEFAULT_SOURCE, null);
+  }
+
+  @Override
+  public void send(String name, double value, @Nullable Long timestamp) throws IOException {
+    addPoint(name, value, timestamp, DEFAULT_SOURCE, null);
+  }
+
+  @Override
+  public void send(String name, double value, @Nullable Long timestamp, String source) throws IOException {
+    addPoint(name, value, timestamp, source, null);
+  }
+
+  @Override
+  public void send(String name, double value, String source, @Nullable Map<String, String> pointTags) throws IOException {
+    addPoint(name, value, null, source, pointTags);
+  }
+
+  @Override
+  public void send(String name, double value, @Nullable Long timestamp, String source,
+                   @Nullable Map<String, String> pointTags) throws IOException {
+    addPoint(name, value, timestamp, source, pointTags);
+  }
+
+  private void addPoint(@NotNull String name, double value, @Nullable Long timestamp, @NotNull String source,
+                        @Nullable Map<String, String> pointTags) throws IOException {
+
+    //TODO: extra validation required?
+    String point = pointToString(name, value, timestamp, source, pointTags);
+    if (point != null) {
+      synchronized(pointsMutex) {
+        this.buffer.add(point);
+      }
+    }
+  }
+
+  private static String escapeQuotes(String raw) {
+    return StringUtils.replace(raw, quote, escapedQuote);
+  }
+
+  @Nullable
+  static String pointToString(String name, double value, @Nullable Long timestamp, String source,
+                               @Nullable Map<String, String> pointTags) {
+
+    if (name == null || name.isEmpty() || source == null || source.isEmpty()) {
+      LOGGER.debug("Invalid point: Empty name/source");
+      return null;
+    }
+
+    if (timestamp == null) {
+      timestamp = System.currentTimeMillis()/1000;
```

Comment:
THis should not be necessary, let's assign the time on the server side upon receipt (eliminate clock drift confusion)